### DISCUSSION TITLE
How can I rewrite nested routes with dynamic parameters?

### DISCUSSION THREAD

**Discussion by alexrequelme** [vercel/next.js#50908] [Help] [answered]
### Summary

I have this config: 
```js
async rewrites() {
    return [
      { source: "/@:username/songs/:songId", destination: "/users/:username/songs/:songId" },
      { source: "/@:username", destination: "/users/:username" },
    ];
  },
```
My question is if I can achieve the same result with just one rule using regex or a similar approach, as I want to avoid adding a separate rule for each nested route.

**Comment by devjiwonchoi** [comment, marked as answer]
This will do the work. You only need to modify the `pathArray`.

```js
const defaultSource = '/@:username'
const defaultDestination = '/users/:username'
const pathArray = ['', '/songs/:songId', ...rest]

// ...

  async rewrites() {
    return pathArray.map((el) => ({
      source: `${defaultSource}${el}`,
      destination: `${defaultDestination}${el}`
    })
  },

// ...
```

**Reply by devjiwonchoi** [reply]
What you are using is a regex itself and the more you want to add the Matcher for rewrites, the more you should write them down is inevitable. 

To simplify your regex, currently it is same as:

```js
source: "/@:slug1/songs/:slug2", destination: "/users/:slug1/songs/:slug2" 
source: "/@:slug1", destination: "/users/:slug1" 
```

As you can see, there is no short-cut regex for your current situation since you do need separate Matchers for different paths.
